src/formatDate.js

```diff
@@ -50,7 +50,35 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const resultedDate = [];
+  const separatorFrom = fromFormat[3];
+  const separatorTo = toFormat[3];
+
+  const separatedDate = date.split((separatorFrom));
+  let currentYear = separatedDate[fromFormat.indexOf('YY')]
+    || separatedDate[fromFormat.indexOf('YYYY')];
+
+  resultedDate[toFormat.indexOf('MM')]
+    = separatedDate[fromFormat.indexOf('MM')];
+
+  resultedDate[toFormat.indexOf('DD')]
+    = separatedDate[fromFormat.indexOf('DD')];
+
+  if (toFormat.includes('YYYY')) {
```

Comment:
`toFormat.includes('YYYY')` can be replaced by comparing the result of `toFormat.indexOf('YYYY')` with -1. So you willn't call an array loop twice (these methods are loops under the hood so are expensive operations). Same for other use cases.